src/effects/lv2/LV2Effect.cpp

```diff
@@ -553,98 +486,77 @@ bool LV2Effect::InitializePlugin()
    AddFeature(LV2_EXTERNAL_UI__Host, &mExternalUIHost);
    AddFeature(LV2_EXTERNAL_UI_DEPRECATED_URI, &mExternalUIHost);
    // Some plugins specify this as a feature
-   AddFeature(LV2_EXTERNAL_UI__Widget, NULL);
-
-   mInstanceAccessFeature = AddFeature(LV2_INSTANCE_ACCESS_URI, NULL);
-   mParentFeature = AddFeature(LV2_UI__parent, NULL);
-
-   AddFeature(NULL, NULL);
-
+   AddFeature(LV2_EXTERNAL_UI__Widget, nullptr);
+   // Two features must be filled in later
+   mInstanceAccessFeature = mFeatures.size();
+   AddFeature(LV2_INSTANCE_ACCESS_URI, nullptr);
+   mParentFeature = mFeatures.size();
+   AddFeature(LV2_UI__parent, nullptr);
    if (!ValidateFeatures(lilv_plugin_get_uri(mPlug)))
-   {
       return false;
-   }
 
-   auto minLength = lilv_world_get(gWorld, lilv_plugin_get_uri(mPlug), node_MinBlockLength, NULL);
-   if (minLength)
-   {
-      if (lilv_node_is_int(minLength))
-      {
-         int val = lilv_node_as_int(minLength);
-         if (mMinBlockSize < val)
-         {
-            mMinBlockSize = val;
-         }
-      }
+   // Adjust the values in the block size features according to the plugin
+   if (auto minLength = lilv_world_get(
+      gWorld, lilv_plugin_get_uri(mPlug), node_MinBlockLength, nullptr)
+      ; lilv_node_is_int(minLength)
+   ){
+      if (auto value = lilv_node_as_int(minLength)
+         ; value > 1
+      )
+         mMinBlockSize = std::max<size_t>(mMinBlockSize, value);
       lilv_node_free(minLength);
    }
-
-   auto maxLength = lilv_world_get(gWorld, lilv_plugin_get_uri(mPlug), node_MaxBlockLength, NULL);
-   if (maxLength)
-   {
-      if (lilv_node_is_int(maxLength))
-      {
-         int val = lilv_node_as_int(maxLength);
-         if (mMaxBlockSize > val)
-         {
-            mMaxBlockSize = val;
-         }
-      }
+   if (auto maxLength = lilv_world_get(
+      gWorld, lilv_plugin_get_uri(mPlug), node_MaxBlockLength, nullptr)
+      ; lilv_node_is_int(maxLength)
+   ){
+      if (auto value = lilv_node_as_int(maxLength)
+         ; value > 1
```

Comment:
ditto

Reply:
I will make it '>= 1'.

I fear what might happen if maximum block size is allowed to be 0.  It could be that processing makes no progress at all.